src/main/TopoMod.F90

```diff
@@ -250,7 +278,14 @@ subroutine UpdateTopo(this, bounds, num_icec, filter_icec, &
     do c = bounds%begc, bounds%endc
        if (.not. this%needs_downscaling_col(c)) then
           g = col%gridcell(c)
-          this%topo_col(c) = atm_topo(g)
+          l = col%landunit(c)
+          if (col%is_hillslope_column(c) .and. downscale_hillslope_meteorology) then
+             this%topo_col(c) = atm_topo(g) &
+                  + (col%hill_elev(c) - mean_hillslope_elevation(l))
+             this%needs_downscaling_col(c) = .true.
+          else
+             this%topo_col(c) = atm_topo(g)
+          endif
```

Comment:
- [ ] The change you made at the end of UpdateTopo is getting closer, but I think it still isn't quite right (sorry for my vague explanation before): I think you need to move the hillslope block of code here (i.e., the conditional block starting with `if (col%is_hillslope_column(c) .and. downscale_hillslope_meteorology) then`) to outside of the `this%needs_downscaling_col(c)` conditional – so that the hillslope adjustments are made to all hillslope columns, whether or not `needs_downscaling_col` is true or false at this point.

- [ ] Also, the comment right above this loop is wrong now: it says that it it shouldn't matter that we set topo_col = atm_topo, but now it DOES matter, because it's needed for the later hillslope adjustment. Can you please fix that comment?

I suggest the following code to address these two things:

```f90
  ! This could operate over a filter like 'allc' in order to just operate over active
  ! points, but I'm not sure that would speed things up much, and would require passing
  ! in this additional filter.
  do c = bounds%begc, bounds%endc
     if (.not. this%needs_downscaling_col(c)) then
        ! For any point that isn't already set to be downscaled, set its topo value to the
        ! atmosphere's topographic height. This is important for the hillslope block
        ! below. For non-hillslope columns, this shouldn't matter, but is useful if
        ! topo_col is written to the history file.
        g = col%gridcell(c)
        this%topo_col(c) = atm_topo(g)
     end if

     if (col%is_hillslope_column(c) .and. downscale_hillslope_meteorology) then
        l = col%landunit(c)
        this%topo_col(c) =  this%topo_col(c) &
             + (col%hill_elev(c) - mean_hillslope_elevation(l))
        this%needs_downscaling_col(c) = .true.
     endif
  end do
```